### DISCUSSION TITLE
CreateRender passing more then one value.

### DISCUSSION THREAD

**Discussion by rgfx** [bryanmylee/svelte-headless-table#74] [Q&A] [answered]
Can I pass more then one value using Create Render. I need it too perform an action. 

```js
        table.column({
            header: ' ',
            accessor: 'user_id',
            cell: ({value, item}) =>
            createRender(GetFilesActions, {user_id: value, name: item.name}),
        }),
```

Doesn't seem to like my code

**Comment by bryanmylee** [comment, marked as answer]
Hi @rgfx, it should work as long as the `GetFilesActions` component has the right prop types. What's the issue you seem to be getting?

As for passing more than one value, are you trying to pass more than one component as the rendered component?

**Reply by rgfx** [reply]
@bryanmylee this is all that is in my component 

```js
<script>
    export let name;
    export let user_id;
</script>
{name}{user_id}
```

this is the error I get
![image](https://user-images.githubusercontent.com/2388078/195246868-c5ba59bb-2eef-4aa6-8549-25af52c670f1.png)

Those values do exist.

**Reply by bryanmylee** [reply]
Oh my bad, I misread the earlier example.

`cell` takes a `DataBodyCell` [as its first argument](https://svelte-headless-table.bryanmylee.com/docs/api/create-columns#columndef-cell-datacell-state-renderconfig).

You're running into an error because `item` does not exist on [`DataBodyCell`](https://svelte-headless-table.bryanmylee.com/docs/api/body-cell#databodycell), so `item.name` throws the error you're getting.

You're probably trying to do something like this:

```ts
table.column({
  header: ' ',
  accessor: 'user_id',
  cell: ({row, value}) =>
    createRender(GetFilesActions, {
      user_id: value, // as accessed with 'user_id'
      name: row.isData() ? row.original.name : '',
    }),
}),
```

A more direct approach would be to pass the whole object as the value, but then you would have to handle sorting on the column appropriately.

```ts
table.column({
  header: ' ',
  // pass the original item as the cell value
  accessor: (item) => item,
  cell: ({value}) =>
    createRender(GetFilesActions, {
      user_id: value.user_id, // `value` is the original item
      name: value.name,
    }),
}),
```

**Reply by rgfx** [reply]
@bryanmylee Thank very much. This what I needed.
